src/grizzly/reduce/strategies/beautifulsoup.py

```diff
@@ -126,3 +130,82 @@ def __iter__(self) -> Generator[list[TestCase]]:
                 LOG.warning("%s failed (reverting)", self.name)
                 lith_tc.dump(file)
             self._current_feedback = None
+
+
+class BeautifulSoupCSSMerge(BeautifulSoupStrategy):
+    name = "css-merge"
+
+    def __iter__(self) -> Generator[list[TestCase]]:
+        """Iterate over potential modifications of testcases according to this strategy.
+
+        The caller should evaluate each testcase set yielded, and call `update` with the
+        result. The caller owns the testcases yielded, and should call `cleanup` for
+        each.
+
+        Yields:
+            Testcases with modifications.
+        """
+        if not self.import_available:
+            LOG.warning("%s not available, skipping strategy.", self.import_name)
+            return
+
+        LOG.info("BeautifulSoup CSS Merge %d files", len(self._files_to_process))
+        for file_no, file in enumerate(self._files_to_process, start=1):
+            LOG.info(
+                "BeautifulSoup CSS Merge %s (file %d/%d)",
+                file.relative_to(self._testcase_root),
+                file_no,
+                len(self._files_to_process),
+            )
+
+            # Use Lithium just to split the file at DDBEGIN/END.
+            # Lithium already has the right logic for DDBEGIN/END.
+            lith_tc = TestcaseLine()
+            lith_tc.load(file)
+
+            data = b"".join(lith_tc.parts)
+
+            style_data: list[str] = []
+            soup = BeautifulSoup(data.decode(), features="html.parser")
+            for tag in soup.find_all():
+                style_attr = tag.attrs.pop("style", None)
+                # collect style data and prepare it to be added to a style tag
+                if isinstance(style_attr, str) and style_attr:
+                    tag_id = tag.get("id")
+                    # use tag id if available otherwise use tag name
+                    if tag_id is not None:
+                        style_data.append(f"#{tag_id} {{")
+                    else:
+                        style_data.append(f"{tag.name} {{")
```

Comment:
Does this happen? It seems like using name will be too vague. I guess you are testing the result anyways.